install.sh

```diff
@@ -472,10 +482,13 @@ do_unmount() {
 do_unmount '/run/k3s'
 do_unmount '/var/lib/rancher/k3s'
 
-nets=$(ip link show | grep 'master cni0' | awk -F': ' '{print $2}' | sed -e 's|@.*||')
-for iface in $nets; do
-    ip link delete $iface;
-done
+# Delete network interface(s) that match 'master cni0'
+while read ignore iface ignore; do
+    iface=${iface%%@*}
+    [ -z "$iface" ] || ip link delete $iface
+done <<-EOF
+    $(ip link show | grep 'master cni0')
+EOF
```

Comment:
Has this change been tested?

Reply:
Thank you so much for spotting the layer 8 issue! Error when merging things together, for sure.

Reply:
Current code on master behavior:

$ ps -o ppid= -o pid=       
13259 13576
13576 13610
$ pid=13259       
$ ps -o ppid= -o pid= | awk "\$1==$pid {print \$2}"
13576
$ pstree $(ps -o ppid= -o pid= | awk "\$1==$pid {print \$2}") 
zsh───pstree
$ 

Today's code for this PR:

$ ps -o ppid= -o pid= | while read parent child; do [ $parent -ne $pid ] || pstree $child; done     
zsh───pstree

(and to show the child pid)

$ ps -o ppid= -o pid= | while read parent child; do [ $parent -ne $pid ] || echo $child; done
13576

Reply:
Thanks for updating! There also appears to be an issue where there are multiple nested EOF here docs. This causes a problem with the installer for me like the following, after this update:

```
$ ./install.sh || echo fail
[INFO]  Finding latest release
[INFO]  Using v0.9.1 as release
[INFO]  Downloading hash https://github.com/rancher/k3s/releases/download/v0.9.1/sha256sum-amd64.txt
[INFO]  Skipping binary downloaded, installed k3s matches hash
[INFO]  Skipping /usr/local/bin/kubectl symlink to k3s, already exists
[INFO]  Skipping /usr/local/bin/crictl symlink to k3s, already exists
[INFO]  Skipping /usr/local/bin/ctr symlink to k3s, command exists in PATH at /usr/bin/ctr
[INFO]  Creating killall script /usr/local/bin/k3s-killall.sh
ip: can't find device 'cni0'
fail
```

Reply:
Yes, things got really weird there. I apologize. Now updated with fix that has been verified (install.sh and then k3s-uninstall.sh) on Raspbian and Ubuntu 64.